include/mapbox/geojsonvt.hpp

```diff
@@ -35,40 +37,90 @@ struct Options {
     uint16_t buffer = 64;
 };
 
-class __attribute__((visibility("default"))) GeoJSONVT {
-public:
-    static const Tile emptyTile;
+namespace detail {
 
-    GeoJSONVT(std::vector<ProjectedFeature> const& features_, const Options& = Options());
+inline uint64_t toID(uint8_t z, uint32_t x, uint32_t y) {
+    return (((1 << z) * y + x) * 32) + z;
+}
 
-    static std::vector<ProjectedFeature> convertFeatures(const std::string& data,
-                                                         const Options& options = Options());
+} // namespace detail
 
-    const Tile& getTile(uint8_t z, uint32_t x, uint32_t y);
+class GeoJSONVT {
+public:
+    const Options options;
 
-    const std::unordered_map<uint64_t, Tile>& getAllTiles() const;
+    GeoJSONVT(const mapbox::geometry::feature_collection<double>& features_,
+              const Options& options_ = Options())
+        : options(options_) {
 
-    // returns the total number of tiles generated until now.
-    uint64_t getTotal() const;
+        const uint32_t z2 = std::pow(2, options.maxZoom);
 
-    const Options options;
+        auto converted = detail::convert(features_, options.tolerance / (z2 * options.extent));
+        auto features = detail::wrap(converted, double(options.buffer) / options.extent);
 
-private:
-    uint8_t splitTile(std::vector<ProjectedFeature> const& features_,
-                      uint8_t z_,
-                      uint32_t x_,
-                      uint32_t y_,
-                      uint8_t cz = 0,
-                      uint32_t cx = 0,
-                      uint32_t cy = 0);
+        splitTile(features, 0, 0, 0);
+    }
+
+    std::map<uint8_t, uint32_t> stats;
+    uint32_t total = 0;
+    std::unordered_map<uint64_t, detail::Tile> tiles;
 
 private:
-    std::unordered_map<uint64_t, Tile> tiles;
-    std::map<uint8_t, uint16_t> stats;
-    uint64_t total = 0;
+    void splitTile(const detail::vt_features& features,
+                   const uint8_t z,
+                   const uint32_t x,
+                   const uint32_t y) {
+
+        if (features.empty())
+            return;
+
+        const double z2 = 1 << z;
+        const uint64_t id = detail::toID(z, x, y);
+
+        const auto it = tiles.find(id);
+
+        if (it == tiles.end()) {
+            const double tolerance =
+                (z == options.maxZoom ? 0 : options.tolerance / (z2 * options.extent));
+
+            tiles.emplace(
+                id, detail::Tile{ features, z, x, y, options.extent, options.buffer, tolerance });
+            stats[z] = (stats.count(z) ? stats[z] + 1 : 1);
+            total++;
+        }
+
+        auto& tile = tiles.find(id)->second;
```

Comment:
This find-or-create-if-missing pattern can be rewritten more efficiently like this:

```
auto it = tiles.find(id);
if (it == tiles.end()) {
    it = tiles.emplace(id, ...).first;
}
auto& tile = it->second;
```

This avoids a second lookup in both cases.